src/connector/src/parser/mysql.rs

```diff
@@ -0,0 +1,156 @@
+};
+use rust_decimal::Decimal as RustDecimal;
+
+pub fn mysql_row_to_datums(mysql_row: &mut MysqlRow, schema: &Schema) -> Vec<Datum> {
+    let mut datums = vec![];
+    for i in 0..schema.fields.len() {
+        let rw_field = &schema.fields[i];
+        let datum = {
+            match rw_field.data_type {
+                DataType::Boolean => {
+                    let v = mysql_row.take::<bool, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Int16 => {
+                    let v = mysql_row.take::<i16, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Int32 => {
+                    let v = mysql_row.take::<i32, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Int64 => {
+                    let v = mysql_row.take::<i64, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Float32 => {
+                    let v = mysql_row.take::<f32, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Float64 => {
+                    let v = mysql_row.take::<f64, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Decimal => {
+                    let v = mysql_row.take::<RustDecimal, _>(i);
+                    v.map(|v| ScalarImpl::from(Decimal::from(v)))
+                }
+                DataType::Varchar => {
+                    // snapshot data doesn't contain offset, just fill None
+                    if rw_field.name.as_str() == OFFSET_COLUMN_NAME {
+                        None
+                    } else {
+                        let v = mysql_row.take::<String, _>(i);
+                        v.map(ScalarImpl::from)
+                    }
+                }
+                DataType::Date => {
+                    let v = mysql_row.take::<NaiveDate, _>(i);
+                    v.map(|v| ScalarImpl::from(Date::from(v)))
+                }
+                DataType::Time => {
+                    let v = mysql_row.take::<chrono::NaiveTime, _>(i);
+                    v.map(|v| ScalarImpl::from(Time::from(v)))
+                }
+                DataType::Timestamp => {
+                    let v = mysql_row.take::<chrono::NaiveDateTime, _>(i);
+                    v.map(|v| ScalarImpl::from(Timestamp::from(v)))
+                }
+                DataType::Timestamptz => {
+                    let v = mysql_row.take::<chrono::NaiveDateTime, _>(i);
+                    v.map(|v| ScalarImpl::from(Timestamptz::from_micros(v.timestamp_micros())))
+                }
+                DataType::Bytea => {
+                    let v = mysql_row.take::<Vec<u8>, _>(i);
+                    v.map(|v| ScalarImpl::from(v.into_boxed_slice()))
+                }
+                DataType::Jsonb => {
+                    let v = mysql_row.take::<serde_json::Value, _>(i);
```

Comment:
When we change to an alternative representation #11805, this would be less efficient because 2 conversions are involved:`&str` -> `serde_json::Value` -> `JsonVal`. It would be more efficient to go directly `&str` -> `JsonVal`. This applies to most usages of `impl From<serde_json::Value> for JsonVal`, which goes beyond this PR.